### PR TITLE
[docs] Initial standard library programmer's guide

### PR DESCRIPTION
**Pull request by milseman**

### CODE REVIEW

File: docs/StandardLibraryProgrammersManual.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Standard Library Programmers Manual
+
+This is meant to be a guide to people working on the standard library. It covers coding standards, code organization, best practices, internal annotations, and provides a guide to standard library internals. This document is inspired by LLVM's excellent [programmer's manual](http://llvm.org/docs/ProgrammersManual.html) and [coding standards](http://llvm.org/docs/CodingStandards.html).
+
+TBD: Should this subsume [StdlibRationals.rst](https://github.com/apple/swift/blob/master/docs/StdlibRationales.rst)?
```
Comment (xwu) [inline docs/StandardLibraryProgrammersManual.md]: Typo in link (rationales, not rationals).

**Comment by milseman** [comment]
(I'm keeping this PR open while I casually flesh out more of the design and/or details. If anyone is interested in starting collaboration in this area before then, please reach out to me)

**Comment by milseman** [comment]
@swift-ci please smoke test

... I guess?

**Comment by milseman** [comment]
@swift-ci please smoke test

**Pull request merged**